atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/SweepStrategy.java

```diff
@@ -29,27 +29,35 @@
 public final class SweepStrategy {
 
     public static final SweepStrategy CONSERVATIVE =
-            new SweepStrategy(Optional.of(SweeperStrategy.CONSERVATIVE), false);
-    public static final SweepStrategy THOROUGH = new SweepStrategy(Optional.of(SweeperStrategy.THOROUGH), true);
+            new SweepStrategy(Optional.of(SweeperStrategy.CONSERVATIVE), false, false);
+    public static final SweepStrategy THOROUGH = new SweepStrategy(Optional.of(SweeperStrategy.THOROUGH), false, true);
 
     private final Optional<SweeperStrategy> sweeperStrategy;
-    private final boolean mustCheckImmutableLockAfterReads;
+    private final boolean mustCheckImmutableLockAfterNonEmptyReads;
+    private final boolean mustCheckImmutableLockAfterEmptyReads;
 
-    private SweepStrategy(Optional<SweeperStrategy> sweeperStrategy, boolean mustCheckImmutableLockAfterReads) {
+    private SweepStrategy(
+            Optional<SweeperStrategy> sweeperStrategy,
+            boolean mustCheckImmutableLockAfterNonEmptyReads,
+            boolean mustCheckImmutableLockAfterEmptyReads) {
         this.sweeperStrategy = sweeperStrategy;
-        this.mustCheckImmutableLockAfterReads = mustCheckImmutableLockAfterReads;
+        this.mustCheckImmutableLockAfterNonEmptyReads = mustCheckImmutableLockAfterNonEmptyReads;
+        this.mustCheckImmutableLockAfterEmptyReads = mustCheckImmutableLockAfterEmptyReads;
     }
 
     public Optional<SweeperStrategy> getSweeperStrategy() {
         return sweeperStrategy;
     }
 
-    public boolean mustCheckImmutableLockAfterReads() {
-        return mustCheckImmutableLockAfterReads;
+    public boolean mustCheckImmutableLock(boolean hasReadEmptyCells) {
+        return hasReadEmptyCells ? mustCheckImmutableLockAfterEmptyReads : mustCheckImmutableLockAfterNonEmptyReads;
     }
 
     public static SweepStrategy from(TableMetadataPersistence.SweepStrategy strategy) {
```

Comment:
We talked about only implementing this change initially for Cassandra backed tables. But the context we usually build an object of this class from is quite annoying to get the KVS implementation. Do people have suggestions on how to do it in a non ugly way?